### PR TITLE
miniflux: init at 2.0.12

### PR DESCRIPTION
**Pull request by benpye**
###### Motivation for this change

Adds Miniflux feed reader server. It's similar to TT-RSS. It would be helpful to have a service config but I have yet to get that far.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/miniflux/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ stdenv
+, lib
```
Comment (dywedir) [inline pkgs/servers/miniflux/default.nix]:
```suggestion
```

**Comment by dywedir** [comment]
Thank you!
Nitpick:
```diff
{ stdenv
- , lib
, buildGoPackage
- , go
, fetchFromGitHub
}:
```

**Comment by dywedir** [comment]
@GrahamcOfBorg build miniflux

**Comment by andir** [comment]
@benpye I have been running miniflux for a while with out of tree modules & packages. Here is the module that I have lying around:

```nix
{ config, lib, pkgs, ...}:
let
  cfg = config.services.miniflux;

  # this is supposed to be used on localhost (in a container) only. The access
  # to the database is only possible from miniflux or whatever else the admin
  # decides to put in the same container..
  dbUser = "miniflux";
  dbPassword = "miniflux";
  dbHost = "localhost";
  dbName = "miniflux";
  pgSuperUser = config.services.postgresql.superUser;
in {
  options = with lib; {
    services.miniflux = {
      enable = mkEnableOption "miniflux";

      package = mkOption {
        type = types.package;
        example = "miniflux";
      };

      user = mkOption {
        type = types.str;
        default = "miniflux";
        example = "miniflux";
        description = ''
          User account the daemon will run under.
        '';
      };

      listenAddress = mkOption {
        type = types.str;
        default = "localhost";
        description = ''
          The host the daemon will listen on.
        '';
      };

      listenPort = mkOption  {
        type = types.int;
        default = 8080;
        example = "8080";
        description = ''
          The port the daemon will listen on.
        '';
      };

      extraConfig = mkOption {
        type = types.lines;
        default = "";
        example = ''
          CREATE_ADMIN=1
          ADMIN_USERNAME=clown
          ADMIN_PASSWORD=s3cret123
        '';
      };

      statePath = mkOption {
        type = types.str;
        default = "/var/miniflux/state";
        description = "miniflux state directory.";
      };
    };
  };

  config = lib.mkIf cfg.enable {
    services.postgresql.enable = lib.mkDefault true;
    users.users."${cfg.user}" = {
      createHome = false;
    };
    systemd.services.miniflux = let
      configFile = pkgs.writeText "miniflux.conf" ''
        RUN_MIGRATIONS=1
        LISTEN_ADDR=${cfg.listenAddress}:${builtins.toString cfg.listenPort}
        DATABASE_URL="postgresql://${dbUser}:${dbPassword}@${dbHost}/${dbName}?sslmode=disable"
        ${cfg.extraConfig}
      '';
    in {
      description = "Miniflux service";
      preStart = ''
        mkdir -p ${cfg.statePath}
        if ! test -e "${cfg.statePath}/db-created"; then
          ${pkgs.sudo}/bin/sudo -u ${pgSuperUser} ${config.services.postgresql.package}/bin/psql postgres -c "CREATE ROLE ${dbUser} WITH LOGIN NOCREATEDB NOCREATEROLE ENCRYPTED PASSWORD '${dbPassword}'"
          ${pkgs.sudo}/bin/sudo -u ${pgSuperUser} ${config.services.postgresql.package}/bin/createdb --owner ${dbUser} ${dbName}
          ${pkgs.sudo}/bin/sudo -u ${pgSuperUser} ${config.services.postgresql.package}/bin/psql ${dbName} -c "CREATE EXTENSION IF NOT EXISTS hstore"
        #  source ${configFile}
        #  ${pkgs.sudo}/bin/sudo -u ${cfg.user} ${cfg.package}/bin/miniflux -migrate
          touch ${cfg.statePath}/db-created
        fi
      '';
      serviceConfig = {
        PermissionsStartOnly = true;
        Type = "simple";
        EnvironmentFile = configFile;
        PrivateTmp = false;
        ExecStart = "${cfg.package}/bin/miniflux.app";
        User = cfg.user;
      };
    };
  };
}
```

It think the migration is done automatically now and thus I commented those lines out.. Not sure anymore. It is a very low priority task on my list....

You probably want to make the constants at the top of the file configurable. Feel free to ping me on a follow-up PR.

**Pull request merged**